File: assignment 4/python/programs/2.py
def highest_ratio():
    N = int(input("Enter the number of students: "))
    print("Height in m and weight in kg")
    data = []
    for i in range(N):
        height = float(input(f"Enter the height of student {i+1}: "))
        weight = float(input(f"Enter the weight of student {i+1}: "))
        data.append((height, weight))
    max_ratio = 0
    for i in range(N):
        height, weight = data[i]
        ratio = weight / height
        if ratio > max_ratio:
            max_ratio = ratio
    return max_ratio


print(highest_ratio())

File: assignment 4/c/programs/2.c
#include <stdio.h>

// Define a structure to store student data
struct student
{
    float height;
    float weight;
};

int main()
{
    int N;
    printf("Enter the number of students: ");
    scanf("%d", &N);
    // Declare an array of structures to store N students
    struct student students[N];
    float ratio, max_ratio = 0;
    for (int i = 0; i < N; i++)
    {
        printf("Enter the height of student %d: ", i + 1);
        scanf("%f", &students[i].height);
        printf("Enter the weight of student %d: ", i + 1);
        scanf("%f", &students[i].weight);
        ratio = students[i].weight / students[i].height;
        if (ratio > max_ratio)
        {
            max_ratio = ratio;
        }
    }
    printf("The highest weight/height ratio is: %f\n", max_ratio);
    return 0;
}
